### PR TITLE
feat: add validation for break/continue statements outside loops

### LINKED ISSUE #378: Missing `continue;` and `break;` statement validation
We don't have the context that it is in a loop to hand for these CST items, so that would need solving too.

Errors returned by Salesforce deploy:

- `Continue statement must be in loop`
- `Break statement must be in loop`

### PR DESCRIPTION
**Pull request by nawforce**
## Summary
- Add validation to ensure `break` and `continue` statements are only used within loop contexts
- Support all loop types: `while`, `do-while`, `for`, and `for-each` loops
- Generate proper error messages matching Salesforce behavior
- Handle edge cases like single-statement while loops

## Implementation Details
- Add `inLoop` flag and `withInLoop` method to `BlockVerifyContext`
- Update loop statements (`ForStatement`, `WhileStatement`, `DoWhileStatement`) to mark their contexts as in-loop
- Add validation logic in `BreakStatement` and `ContinueStatement` classes
- Use mutation-based approach for `while`/`do-while` to handle single statements safely

## Test Coverage
- 23 comprehensive tests covering all scenarios
- Invalid usage: break/continue outside loops (methods, if blocks, switch, try-catch)
- Valid usage: all loop types including nested loops and complex control structures
- Edge cases: while loops with single statements (no blocks)

## Error Messages
- `"Break statement must be in loop"`
- `"Continue statement must be in loop"`

Resolves #378

### CODE REVIEW

File: jvm/src/main/scala/com/nawforce/apexlink/cst/VerifyContext.scala [hunk 1]
```diff
@@ -388,6 +390,25 @@ abstract class BlockVerifyContext(parentContext: VerifyContext)
...
+  /** Temporarily set the loop flag for the duration of an operation */
+  def withInLoop[T](op: => T): T = {
+    val previousInLoop = inLoop
+    this.inLoop = true
+    try {
+      op
+    } finally {
+      this.inLoop = previousInLoop
+    }
+  }
```
Comment (pwrightcertinia) [inline jvm/src/main/scala/com/nawforce/apexlink/cst/VerifyContext.scala:410]: When would it be safe to set the flag on the loop context and leave it ?

My understanding is we don't always create a separate context in the loop verify - therefore we don't want to poison that parent context by setting true loop flag on it. Can we create contexts in the while statements to make a permanent flag work?

**Review by pwrightcertinia — APPROVED** [review]
